tachyon/math/finite_fields/prime_field_fallback.h

```diff
@@ -277,17 +277,21 @@ class PrimeField<_Config, std::enable_if_t<!_Config::kUseAsm &&
   }
 
   // MultiplicativeGroup methods
-  constexpr PrimeField Inverse() const {
+  constexpr std::optional<PrimeField> Inverse() const {
     PrimeField ret;
-    ret.value_ = value_.template MontgomeryInverse<Config::kModulusHasSpareBit>(
-        Config::kModulus, Config::kMontgomeryR2);
+    if (UNLIKELY(
+            !value_.template MontgomeryInverse<Config::kModulusHasSpareBit>(
+                Config::kModulus, Config::kMontgomeryR2, ret.value_)))
+      return std::nullopt;
```

Comment:
Do you think we have to add logging here? If you agree, please add logging to all the `InverseXXX()` methods.